### PR TITLE
feat: add binary sensor for connectivity status

### PR DESCRIPTION
**Pull request by CFenner**
**Work in Progress**

Adds a connectivity binary sensor as for ESPhome devices. Related to [#54](https://github.com/hellqvio86/home_assistant_casambi/issues/54#issuecomment-1321535411). Requires #74.

Bildschirm­foto 2022-11-23 um 22 32 29

### CODE REVIEW

**Review by hellqvio86 — APPROVED** [review]
Lgtm, thanks for the the PR 🙂

**Comment by CFenner** [comment]
Connectivity Sensor works now, but light entity still stays available and is switchable without effect.

Bildschirm­foto 2022-11-27 um 23 42 59

File: custom_components/casambi/casambi/CasambiEntity.py [hunk 1]
```diff
@@ -66,3 +57,7 @@ def available(self) -> bool:
     def should_poll(self):
         """Disable polling by returning False"""
         return False
+
+    def __repr__(self) -> str:
+        """Return the representation."""
+        return f"<Casambi {self.unit.name}: unit={self.unit}"
```
Comment (CFenner) [inline custom_components/casambi/casambi/CasambiEntity.py:63]:
```suggestion
        return f"<Casambi {self.unit.name}: unit={self.unit}>"
```

File: custom_components/casambi/casambi/CasambiOverheatBinarySensorEntity.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        _LOGGER.debug(f"Casambi overheat binary sensor - init - start")
+
+        CasambiBinarySensorEntity.__init__(self, unit, controller, hass,
+            "Overheat", BinarySensorDeviceClass.HEAT)
+
+        _LOGGER.debug(f"Casambi overheat binary sensor - init - end")
+
+    def __repr__(self) -> str:
+        """Return the representation."""
+        return f"<Casambi overheat binary sensor {self.unit.name}: unit={self.unit}"
```
Comment (CFenner) [inline custom_components/casambi/casambi/CasambiOverheatBinarySensorEntity.py:20]:
```suggestion
        return f"<Casambi overheat binary sensor {self.unit.name}: unit={self.unit}>"
```

File: custom_components/casambi/casambi/CasambiStatusBinarySensorEntity.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        """Process callback message, update home assistant light state"""
+        _LOGGER.debug(f"process_update: self: {self} data: {data}")
+        if self.enabled:
+            if self._unit_unique_id in data:
+                # Device needs to be enabled for us to schedule updates
+                self.async_schedule_update_ha_state(True)
+
+    def __repr__(self) -> str:
+        """Return the representation."""
+        return f"<Casambi status binary sensor {self.unit.name}: unit={self.unit}"
```
Comment (CFenner) [inline custom_components/casambi/casambi/CasambiStatusBinarySensorEntity.py:51]:
```suggestion
        return f"<Casambi status binary sensor {self.unit.name}: unit={self.unit}>"
```

File: custom_components/casambi/casambi/CasambiStatusBinarySensorEntity.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        _LOGGER.debug(f"Casambi status binary sensor - init - end")
+
+    @property
+    def entity_category(self):
+        return EntityCategory.DIAGNOSTIC
+
+    @property
+    def available(self) -> bool:
+        """Connectivity entity is always availble."""
```
Comment (CFenner) [inline custom_components/casambi/casambi/CasambiStatusBinarySensorEntity.py:27]:
```suggestion
        """Connectivity entity is always available."""
```

File: custom_components/casambi/binary_sensor.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    return True
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities):
+    _LOGGER.debug("Setting up binary sensor entities.")
+    # config = hass.data[DOMAIN][config_entry.entry_id]
+
+    # TODO: extract casambi controller from light entity
+    casambi_controller = hass.data[DOMAIN][CONF_CONTROLLER]
+
+    units = casambi_controller.controller.get_units()
```
Comment (CFenner) [inline custom_components/casambi/binary_sensor.py:25]:
```suggestion
    units = casambi_controller.aiocasambi_controller.get_units()
```

**Review by hellqvio86 — APPROVED** [review]
lgtm, thanks for the pull request :slightly_smiling_face:

**Pull request merged**